### PR TITLE
Add result class unit tests

### PR DESCRIPTION
**Pull request by alfinkel**
## What:

Add unit tests to ensure that the Result class logic is tested thoroughly

## How:

- Made a create_views() helper test function available to all test subclasses of UnitTestDbBase
- Added tests for ResultException
- Added tests for Result data access via index value
- Added tests for Result data access via key value
- Added tests for Result data access via index slicing
- Added tests for Result data access via key slicing
- Added tests for Result data access via iteration

## Testing:

- Added tests for ResultException
- Added tests for Result data access via index value
- Added tests for Result data access via key value
- Added tests for Result data access via index slicing
- Added tests for Result data access via key slicing
- Added tests for Result data access via iteration

## Reviewers:

reviewer: @rhyshort 
reviewer: @emlaver 

## Issues:

- #121

### CODE REVIEW

**Comment by rhyshort** [comment]
+1

File: tests/unit/db/result_tests.py [hunk 1]
```diff
@@ -88,43 +106,522 @@ def test_constructor(self):
...
+        self.assertEqual(cm.exception.status_code, 101)
+
+        with self.assertRaises(ResultException) as cm:
+            invalid_result = result[5: 2]
+        self.assertEqual(cm.exception.status_code, 101)
+
+    def test_get_item_index_slice_using_start_stop(self):
+        """
+        Test getting an index slice by using start and stop slice values.
+        """
```
Comment (emlaver) [inline tests/unit/db/result_tests.py]: Is it worth having a test with the same start/stop value? i.e. result[98:98] or result[0:0]

File: tests/unit/db/result_tests.py [hunk 2]
```diff
@@ -88,43 +106,522 @@ def test_constructor(self):
...
+        with self.assertRaises(ResultException) as cm:
+            invalid_result = result['foo': ['bar', 'baz']]
+        self.assertEqual(cm.exception.status_code, 101)
+
+        ten = ResultByKey(10)
+        with self.assertRaises(ResultException) as cm:
+            invalid_result = result['foo': ten]
+        self.assertEqual(cm.exception.status_code, 101)
+
+    def test_get_item_key_slice_using_start_stop(self):
```
Comment (emlaver) [inline tests/unit/db/result_tests.py]: Is it worth having a test with the same start/stop key slice?
i.e. `result[ResultByKey('julia097'): ResultByKey('julia097')]`
or
 `result[['julia', 97]: ['julia', 97]]`

File: tests/unit/db/result_tests.py (hunk 1 above)
Comment (alfinkel) [inline tests/unit/db/result_tests.py]: Good catch.  I updated the code to raise an invalid argument exception in this case and added a test.  Done in 033b46ea1539fa63309b68ca11b7da411ee15744 and e6d1b8b.

File: tests/unit/db/result_tests.py (hunk 2 above)
Comment (alfinkel) [inline tests/unit/db/result_tests.py]: Since this logic works as expected, I added tests to confirm the functionality.  Done in e6d1b8b.

**Comment by emlaver** [comment]
+1

**Pull request merged**